Azure/communication-ui-library-ios
# [Feature] Supporting pinch, zoom, pan gestures and "double tap to zoom" in video screen share view in a call

## Purpose
In this PR I drafted the feature supporting user pinch, zoom and pan gestures in video screen share in a call. The functionalities resembles the MS Teams application UX.
Additional features introduced:
* double tap to zoom in / zoom out
* only zooming the actual video content, not the entire rendered view from SDK
* Improved screen share video quality in iPhone.


https://user-images.githubusercontent.com/90668345/149402996-ff02c754-88ae-4d9a-b1dd-4bacf8c96a94.mov


## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code
* Join a call either using group or Teams call, in iOS app and Teams application (desktop)
* Turn on screenshare in Teams application (desktop)
* Start to perform pinch, zoom and pan gesture on the screen share video 
* Follow **What to Check section**
```

## What to Check
Verify that the following are valid
* Check if the pinch, zoom and pan gestures works in screen share video, in all supported orientations during a call
* Check if double tap to zoom in / out is working as well.

## Other Information
